go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/hashicorp/raft
 
-go 1.16
+go 1.20
```

Comment:
Is this required? I don't think we want to force all downstream users of the library to only be able to build on the latest go release.

This number in go.mod is the "minimum supported version" not "they current version" and we should only update it if we are actually using a new feature of the language that requires at least this version.

If we made this change, we'd could have issues with our product builds if they are not yet ready for 1.20 or even have trouble pulling fixes into backports of our old releases that have to use older go versions etc.

Reply:
I think last time we bumped it we decided "last two" on the basis that that covered most of our supported product releases. But IMO we should leave this as 1.16 until we actually have a specific reason to need language features that are worth breaking compatibility for.

Might be worth checking what Go versions last supported Consul/Nomad/Vault branches built from of you want to be sure but IMO risk is low for this library that we'd introduce backward incompatible changes given how little we commit here so last 2 versions is probably fine?

Reply:
Look like Consul/Nomad is on 1.20 and Vault is on 1.19. I will continue building on latest 2 Go versions, but will revert the changes to `go.mod` even though it's "probably" safe.